resources/ansible/upgrade_uploaders.yml

```diff
@@ -0,0 +1,41 @@
+---
+- name: upgrade the uploaders
+  hosts: all
+  become: True
+  vars:
+    safe_version: "{{ safe_version }}"
+    safe_archive_url: "https://sn-cli.s3.eu-west-2.amazonaws.com/safe-{{ safe_version }}-x86_64-unknown-linux-musl.tar.gz"
+  tasks:
+    - name: stop the uploader service
+      systemd:
+        name: safe_uploader
+        state: stopped
```

Comment:
If the uploader service is stopped and restarted.  Is a new file (delete/create) takes place on the uploader.csv or its always an `>> append` operation from shell script executed by the service/wrapper?

If a new reset of the file, you would need to stop/start telegraf possibly in this upgrade workflow as it would have a handle to the old file on disk, and it may break its tail operation causing a permanent outage to uploader metrics.